lib/waiter.js

```diff
@@ -0,0 +1,59 @@
+class Waiter {
+  constructor (promise, cb, expectReject) {
```

Comment:
The `expectReject` argument is added compared to current lib/test.js so the constructor can just set the correct value in the first place.